src/components/Spacing.js

```diff
@@ -1,53 +1,58 @@
 import styled from "styled-components";
 import PropTypes from "prop-types";
 import spacing from "../theme/spacing";
-import { mediumAndUp, largeAndUp, smallAndUp } from "../theme/mediaQueries";
+import {
+  mediumAndUp,
+  largeAndUp,
+  smallAndUp,
+  xLargeAndUp
+} from "../theme/mediaQueries";
 
 const getSpacing = val => spacing[val] || val || 0;
 
 const Spacing = styled.div`
   width: 100%;
-  padding-top: ${props => getSpacing(props.top.xSmall)};
-  padding-bottom: ${props => getSpacing(props.bottom.xSmall)};
+  padding-top: ${({ top }) => getSpacing(top.xSmall)};
+  padding-bottom: ${({ bottom }) => getSpacing(bottom.xSmall)};
 
   ${smallAndUp`
-    padding-top: ${props => getSpacing(props.top.small || props.top.xSmall)};
-    padding-bottom: ${props =>
-      getSpacing(props.bottom.small || props.bottom.xSmall)};
+    padding-top: ${({ top }) => getSpacing(top.small || top.xSmall)};
+    padding-bottom: ${({ bottom }) =>
+      getSpacing(bottom.small || bottom.xSmall)};
   `};
 
   ${mediumAndUp`
-    padding-top: ${props =>
-      getSpacing(props.top.medium || props.top.small || props.top.xSmall)};
-    padding-bottom: ${props =>
-      getSpacing(
-        props.bottom.medium || props.bottom.small || props.bottom.xSmall
-      )};
+    padding-top: ${({ top }) =>
+      getSpacing(top.medium || top.small || top.xSmall)};
+    padding-bottom: ${({ bottom }) =>
+      getSpacing(bottom.medium || bottom.small || bottom.xSmall)};
   `};
 
   ${largeAndUp`
-    padding-top: ${props =>
+    padding-top: ${({ top }) =>
+      getSpacing(top.large || top.medium || top.small || top.xSmall)};
+    padding-bottom: ${({ bottom }) =>
       getSpacing(
-        props.top.large ||
-          props.top.medium ||
-          props.top.small ||
-          props.top.xSmall
+        bottom.large || bottom.medium || bottom.small || bottom.xSmall
       )};
-    padding-bottom: ${props =>
+  `};
+
+  ${xLargeAndUp`
```

Comment:
I think we should be prioritizing `top.xLarge` and `bottom.xLarge` over these other size properties at `xLargeAndUp`, right?

Reply:
Oh yes. I think I last it somewhere on my way :) Fixed.

Reply:
Awesome, thanks!